test/commands/provar/automation/metadata/download.nut.ts

```diff
@@ -1,194 +1,332 @@
-    );
-    // download metadata for the connection
+  it('Metadata should not be downloaded for the invalid user name and return the error', () => {
     const result = execCmd<SfProvarCommandResult>(
-      `${commandConstants.SF_PROVAR_AUTOMATION_METADATA_DOWNLOAD_COMMAND} -c RegressionOrg`
+      `${commandConstants.SF_PROVAR_AUTOMATION_METADATA_DOWNLOAD_COMMAND} -c PrereleaseOrg`
     ).shellOutput;
     expect(result.stderr).to.include(DOWNLOAD_ERROR);
   });
 
-  it('Metadata should not be downloaded as provarHome & projectPath are not correct and return the error in json format', () => {
+  it('Metadata should not be downloaded for the invalid user name and return the error in json format', () => {
     const result = execCmd<SfProvarCommandResult>(
-      `${commandConstants.SF_PROVAR_AUTOMATION_METADATA_DOWNLOAD_COMMAND} -c RegressionOrg --json`
+      `${commandConstants.SF_PROVAR_AUTOMATION_METADATA_DOWNLOAD_COMMAND} -c PrereleaseOrg --json`
     ).jsonOutput;
     expect(result?.result.success).to.deep.equal(false);
     // eslint-disable-next-line
     expect((result?.result.errors?.[0] as any)?.code).to.equals('DOWNLOAD_ERROR');
   });
 
-  it('Metadata should be downloaded for the provided connection and return the success message', () => {
-    const SET_PROVAR_HOME_VALUE = '"./ProvarHome"';
-    const SET_PROJECT_PATH_VALUE = '"./ProvarRegression/AutomationRevamp"';
-    // set provarHome and projectPath locations
-    execCmd<SfProvarCommandResult>(
-      `${commandConstants.SF_PROVAR_AUTOMATION_CONFIG_SET_COMMAND} "provarHome"=${SET_PROVAR_HOME_VALUE}`
-    );
-    execCmd<SfProvarCommandResult>(
-      `${commandConstants.SF_PROVAR_AUTOMATION_CONFIG_SET_COMMAND} "projectPath"=${SET_PROJECT_PATH_VALUE}`
-    );
+  it('Metadata should not be downloaded when user does not have download permissions and return the error', () => {
+    const result = execCmd<SfProvarCommandResult>(
+      `${commandConstants.SF_PROVAR_AUTOMATION_METADATA_DOWNLOAD_COMMAND} -c NonAdmin`
+    ).shellOutput;
+    expect(result.stderr).to.include(DOWNLOAD_ERROR);
+  });
+
+  it('Metadata should not be downloaded as provarHome & projectPath are not correct and return the error message', async () => {
+    const ProvarDXPropertiesFilePath = path.join(process.cwd(), './provardx-properties.json');
+
+    // Read the config.json file
+    const ProvarDXPropertiesFileData = await fileSystem.readFile(ProvarDXPropertiesFilePath, { encoding: 'utf8' });
+
+    // Parse the JSON data
+    const ProvarDXPropertiesFileParsed = JSON.parse(ProvarDXPropertiesFileData);
+    ProvarDXPropertiesFileParsed['provarHome'] = './ProvarHome1';
+
+    // Convert the updated JSON object back to a string
+    const updatedPropertiesFileData = JSON.stringify(ProvarDXPropertiesFileParsed, null, 4);
+
+    // Write the updated JSON string back to the config.json file
+    await fileSystem.writeFile(ProvarDXPropertiesFilePath, updatedPropertiesFileData, 'utf8');
```

Comment:
I think it is there in every NUT
If we are doing these operations at multiple places that is at multiple NUTs , either we can have it in before() method if used in mostly all NUTs or can derive a method and call it at every place.

Reply:
It is not in every NUT and also wherever this type of line present , this is updating the boilerplate value for different key for negative scenario checking.